core/src/helpers/extractEventFromTransaction.ts

```diff
@@ -0,0 +1,18 @@
+import { ethers } from 'ethers';
+
+const extractEventFromTransaction = async (
+    transaction: ethers.ContractTransaction,
+    eventSignature: string,
+    contractInterface: ethers.utils.Interface
+) => {
+    const receipt = await transaction.wait();
+    const eventLogs = receipt.logs.filter(log => log.topics[0] === ethers.utils.id(eventSignature));
+    const filteredLogs = eventLogs.map(log => ({ topics: log.topics, data: log.data }));
+    const events = filteredLogs.map(log => contractInterface.parseLog(log));
+    if (events.length !== 1) {
```

Comment:
I don't think it's anyhow restricted by the EVM. In fact, I would expect multiply `Transfer` events in one big (eg swap) transaction.
```suggestion
    if (events.length < 1 ) {
```